agent/api/task/task.go

```diff
@@ -846,6 +876,81 @@ func (task *Task) dockerHostConfig(container *apicontainer.Container, dockerCont
 	return hostConfig, nil
 }
 
+func (task *Task) awslogsEndpointOverride(hostConfig *docker.HostConfig, dockerVersion string) bool {
+	if hostConfig.LogConfig.Type != awslogsType {
+		return false
+	}
+
+	isVersionMatch := false
+	if task.isDockerCEVersion(dockerVersion) {
+		match, err := utils.Version(dockerVersion).Matches(">=" + awslogsEndpointMinimumDockerCEVersion)
+		if err != nil {
+			seelog.Warnf("Unable to compare docker version: %v", err)
+		} else if match {
+			isVersionMatch = true
+		}
+	} else if task.isDockerEEVersion(dockerVersion) {
+		match, err := utils.Version(dockerVersion).Matches(">=" + awslogsEndpointMinimumDockerEEVersion)
+		if err != nil {
+			seelog.Warnf("Unable to compare docker version: %v", err)
+		} else if match {
+			isVersionMatch = true
+		}
+	} else {
+		seelog.Warnf("Docker version %s is neither CE version or EE version", dockerVersion)
```

Comment:
This shouldn't be a warning.  Older versions of Docker won't be CE or EE, and customers that are building their own versions may be setting different things here too.